src/nupic/algorithms/KNNClassifier.py

```diff
@@ -845,31 +869,35 @@ def getPartitionId(self, i):
 
   def getPartitionIdPerPattern(self):
     """
-    Returns a list of numPatterns elements where the i'th position contains
-    the integer partition Id associated with pattern i. If pattern i had no
-    partition Id, it's value will be numpy.inf
+    Gets all the partition ids.
+
+    TODO: This method doesn't make sense to me. --Matt
+
+    :returns: a list of numPatterns elements where the i'th position contains
+        the integer partition Id associated with pattern i. If pattern i had no
+        partition Id, it's value will be numpy.inf
```

Comment:
@marionleborgne Does this method make sense to you? It says it accepts `i` and returns one id, but it actually returns all of them. I'll see if it is actually used anywhere, but if not I'll probably remove it. It's false advertising.